compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -1470,8 +1479,8 @@ trait Implicits:
                     // need a candidate better than the two ambiguous alternatives.
                     val ambi = fail.reason.asInstanceOf[AmbiguousImplicits]
                     healAmbiguous(fail, newCand =>
-                      compareAlternatives(newCand, ambi.alt1) > 0 &&
-                      compareAlternatives(newCand, ambi.alt2) > 0)
+                      compareAlternatives(newCand, ambi.alt1, disambiguate = true) > 0 &&
+                      compareAlternatives(newCand, ambi.alt2, disambiguate = true) > 0)
```

Comment:
I'm not sure we can add this here, we needed to `compareAlternative` to only be called with `disambiguate := true` when the two alternatives is has been called with have already been checked to be successful implicit candidates.
Otherwise, we way report incorrect warnings of change in resolution, since the alternative way be found to be un-applicable later on.